app/lib/katello/resources/registry.rb

```diff
@@ -44,9 +44,9 @@ def load_class
             uri = URI.parse(registry_url)
             self.prefix = uri.path
             self.site = "#{uri.scheme}://#{uri.host}:#{uri.port}"
-            self.ssl_client_cert = ::Cert::Certs.ssl_client_cert
-            self.ssl_client_key = ::Cert::Certs.ssl_client_key
             self.ca_cert_file = ca_cert_file
+            pulp_primary.pulp3_ssl_configuration(self) if pulp_primary
```

Comment:
If not, should it fall back to the previous config? Should this actually happen at line 36?

Reply:
Fair question, I think I would need @ianballou thoughts here. There is a test where `pulp_primary` does not exist, and thus the cause of the `if` check here. That's either by design or something the test is doing just to be 'simple'. Can we have a registry resource without a smart-proxy? I think not, but let's see what is said.

Reply:
I can't see a case either when we would have a registry resource without a smart proxy.  Katello would be very broken without a `pulp_primary` smart proxy.

I have a PR coming that will give all tests a `pulp_primary`.  I'm almost certain the test you're referring to doesn't have a `pulp_primary` simply because there isn't one by default at the moment. Feel free to create one in the test to make it pass.